### PR TITLE
Add a new Tree.adjust() method for handling rules

### PR DESCRIPTION
**Pull request by psss**
Implement the new method for handling context rules.
Add unit test coverage for the new functionality.
Fix inheritance to work for dictionary init as well.

### CODE REVIEW

File: fmf/base.py [hunk 1]
```diff
@@ -234,6 +245,89 @@ def update(self, data):
         log.debug("Data for '{0}' updated.".format(self))
         log.data(pretty(self.data))
 
+    def adjust(self, context, key='adjust', undecided='skip'):
+        """
+        Adjust tree data based on provided context and rules
+
+        The 'context' should be an instance of the fmf.context.Context
```
[the remark this answers is not in the document]
Comment (lukaszachy) [reply fmf/base.py:252]: Done in HEAD of pull/83

**Comment by psss** [comment]
Rebased on the merged `Context` implementation and adjusted naming. Ready for review.

**Review by lukaszachy — APPROVED** [review]
Nice, I've tried it and it just-works. LGMT

**Comment by psss** [comment]
Great, thanks for the review. Merging.

**Pull request merged**